include/fmt/printf.h

```diff
@@ -399,10 +399,10 @@ template <typename OutputIt, typename Char>
 typename basic_printf_context<OutputIt, Char>::format_arg
 basic_printf_context<OutputIt, Char>::get_arg(unsigned arg_index) {
   if (arg_index == internal::max_value<unsigned>())
-    arg_index = parse_ctx_.next_arg_id();
+    arg_index = static_cast<unsigned>(parse_ctx_.next_arg_id());
```

Comment:
Please change `arg_index` to `int` and remove all the casts.

Reply:
This will be a bit tricky because of line 401. Should I change the sentinel value to internal::max_value<int>() ?

Reply:
I think sentinel can be changed to -1 once `arg_index` is signed.

Reply:
I was thinking of wrapping the sentinel value in a constant, introduced via an enum so it is portable to older compilers.